### PR TITLE
curl handling fixes

### PR DESCRIPTION
**Pull request by rwos**
* (merged from upstream master) curl_reset (resets curl options) before each request
* curl_close/curl_init only every $X requests (1000 by default)
* add possibility to log curl_getinfo output

### CODE REVIEW

File: src/Sag.php [hunk 1]
```diff
@@ -782,7 +782,7 @@ public function replicate($src, $target, $continuous = false, $createTarget = nu
       $data->filter = $filter;
 
       if($filterQueryParams) {
-        $data->filterQueryParams = $filterQueryParams;
+        $data->query_params = $filterQueryParams;
```
Comment (rwos) [inline src/Sag.php]: this came in with the upstream master merge

File: src/SagUserUtils.php [hunk 2]
```diff
@@ -1,178 +0,0 @@
-<?php
```
Comment (rwos) [inline src/SagUserUtils.php]: also apparently deleted by upstream

File: src/httpAdapters/SagCURLHTTPAdapter.php [hunk 3]
```diff
@@ -101,9 +108,14 @@ public function procPacket($method, $url, $data = null, $reqHeaders = array(), $
       }
     }
 
+    curl_reset($this->ch);
```
Comment (rwos) [inline src/httpAdapters/SagCURLHTTPAdapter.php]: upstream change (rest of the diff of this file are my hacks)

**Comment by rwos** [comment]
never mind, back to me for now

File: src/httpAdapters/SagCURLHTTPAdapter.php [hunk 4]
```diff
@@ -36,12 +41,24 @@ public function __destruct()
...
+  }
+
   public function procPacket($method, $url, $data = null, $reqHeaders = array(), $specialHost = null, $specialPort = null) {
 
-    // destroy and recreate curl handle because otherwise weird things
-    // happen if you make multiple requests with the same connection
-    curl_close($this->ch);
-    $this->ch = curl_init();
+    $this->curlCalls += 1;
+    if ($this->curlCalls >= $this->maxCurlCalls) {
```
Comment (bluemaex) [inline src/httpAdapters/SagCURLHTTPAdapter.php]: is this a long running process or how does this work? If not this is probably reset on every call?

Comment (rwos) [inline src/httpAdapters/SagCURLHTTPAdapter.php]: @bluemaex yes, it's long-running, only constructed once. We used to recycle curl for every request, but now we have upstream's curl_reset() in it (which fixes the bug for which we originally did the recycle) and @baccenfutter suggested recycling only every $X requests. So I did that.

Comment (rwos) [inline src/httpAdapters/SagCURLHTTPAdapter.php]: @bluemaex ...long-running *in the rabbitMQ consumers* - the app uses this too and there it's obviously just constructed and then thrown away.

**Comment by bluemaex** [comment]
looks gut to me :+1: :rocket:

**Pull request merged**